pkg/ingester/wal.go

```diff
@@ -413,16 +413,13 @@ func lastCheckpoint(dir string) (string, int, error) {
 	for i := 0; i < len(dirs); i++ {
 		di := dirs[i]
 
-		if !strings.HasPrefix(di.Name(), checkpointPrefix) {
+		idx, err := checkpointIndex(di.Name())
```

Comment:
This would consider a temporary checkpoint as a valid checkpoint for the last checkpoint, which is wrong. We should be skipping them here.

Reply:
:+1: Good catch, thanks! I made `checkpointIndex` take another arg as to whether temporary checkpoints should be included or not.